File: src/AccelByte.Extend.Vivox.Authentication.Server/Vivox/VivoxConfigProvider.cs
// Copyright (c) 2024 AccelByte Inc. All Rights Reserved.
// This is licensed software from AccelByte Inc, for limitations
// and restrictions contact your company contract manager.

using System;

namespace AccelByte.Extend.Vivox.Authentication.Server.Vivox
{
    public class VivoxConfigProvider : IVivoxConfigProvider
    {
        public string Issuer { get; set; } = "demo";

        public string Domain { get; set; } = "tla.vivox.com";

        public string SigningKey { get; set; } = "";

        public string ChannelPrefix { get; set; } = "confctl";

        public string Protocol { get; set; } = "sip";

        public int DefaultExpiry { get; set; } = 90;

        public void ReadEnvironmentVariables()
        {
            string? vIssuer = Environment.GetEnvironmentVariable("VIVOX_ISSUER");
            if ((vIssuer != null) && (vIssuer.Trim() != ""))
                Issuer = vIssuer.Trim();

            string? vDomain = Environment.GetEnvironmentVariable("VIVOX_DOMAIN");
            if ((vDomain != null) && (vDomain.Trim() != ""))
                Domain = vDomain.Trim();

            string? vSigningKey = Environment.GetEnvironmentVariable("VIVOX_SIGNING_KEY");
            if ((vSigningKey != null) && (vSigningKey.Trim() != ""))
                SigningKey = vSigningKey.Trim();
        }
    }
}

File: src/main/java/net/accelbyte/extend/serviceextension/vivox/VivoxConfigProvider.java
package net.accelbyte.extend.serviceextension.vivox;

import lombok.Data;

@Data
public class VivoxConfigProvider implements ConfigProvider {
    public String issuer = "demo";;
    public String domain = "tla.vivox.com";
    public String signingKey = "";
    public String channelPrefix = "confctl";
    public String protocol = "sip";;
    public int defaultExpiry = 90;

    public void readEnvironmentVariables()
    {
        String vIssuer = System.getenv("VIVOX_ISSUER");
        if ((vIssuer != null) && (vIssuer.trim() != ""))
            issuer = vIssuer.trim();

        String vDomain = System.getenv("VIVOX_DOMAIN");
        if ((vDomain != null) && (vDomain.trim() != ""))
            domain = vDomain.trim();

        String vSigningKey = System.getenv("VIVOX_SIGNING_KEY");
        if ((vSigningKey != null) && (vSigningKey.trim() != ""))
            signingKey = vSigningKey.trim();
    }
}
